neural_compressor/common/base_config.py

```diff
@@ -40,30 +40,71 @@
 
 
 # Dictionary to store registered configurations
-registered_configs = {}
 
 
-def register_config(framework_name="None", algo_name=None):
+class ConfigRegistry:
+    registered_configs = {}
+
+    @classmethod
+    def register_config(cls, framework_name="None", algo_name=None, priority=0):
```

Comment:
How about renaming the register_config to register_config_impl to distinguish the API and implementation?
```python
@register_config(...)  # API
Class XxxConfig:
      ....

```